src/common/io_util.cc

```diff
@@ -131,57 +97,67 @@ Status SockSetTcpKeepalive(int fd, int interval) {
 }
 
 Status SockConnect(const std::string &host, uint32_t port, int *fd, int conn_timeout, int timeout) {
-  if (conn_timeout == 0) {
-    auto s = SockConnect(host, port, fd);
-    if (!s) return s;
-  } else {
-    *fd = socket(AF_INET, SOCK_STREAM, 0);
-    if (*fd == NullFD) return Status::FromErrno();
-  }
-
-  auto exit = MakeScopeExit([fd] {
-    close(*fd);
-    *fd = NullFD;
-  });
+  addrinfo hints, *servinfo = nullptr, *p = nullptr;
 
-  if (conn_timeout != 0) {
-    sockaddr_in sin{};
-    sin.sin_family = AF_INET;
-    sin.sin_addr.s_addr = inet_addr(host.c_str());
-    sin.sin_port = htons(port);
+  memset(&hints, 0, sizeof(hints));
+  hints.ai_family = AF_UNSPEC;
+  hints.ai_socktype = SOCK_STREAM;
 
-    fcntl(*fd, F_SETFL, O_NONBLOCK);
-    connect(*fd, reinterpret_cast<sockaddr *>(&sin), sizeof(sin));
+  if (int rv = getaddrinfo(host.c_str(), std::to_string(port).c_str(), &hints, &servinfo); rv != 0) {
+    return {Status::NotOK, gai_strerror(rv)};
+  }
+  if (int rv = getaddrinfo(host.c_str(), std::to_string(port).c_str(), &hints, &servinfo); rv != 0) {
+    return {Status::NotOK, gai_strerror(rv)};
+  }
```

Comment:
Duplicated?

Reply:
Yes, it's duplicated code. I will revise this PR again and add a test case for this issue.